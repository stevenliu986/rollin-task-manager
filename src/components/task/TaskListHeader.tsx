const TaskListHeader = () => {
    return (
        <div style={{
            width: '100%',
            height: '53px',
            display: 'flex',
            justifyContent: 'space-evenly',
            alignItems: 'center',
            backgroundColor: '#252525',
            borderTopLeftRadius: '16px',
            borderTopRightRadius: '16px',
        }}>
            <div
                style={{
                    padding: '0 25px',
                    color: '#FFF',
                    position: 'relative',
                    cursor: 'pointer'
                }}
            >
                In Progress
            </div>

            <div
                style={{
                    padding: '0 25px',
                    color: '#FFF',
                    position: 'relative',
                    cursor: 'pointer'
                }}
            >
                Completed
            </div>
        </div>
    );
};

export default TaskListHeader;