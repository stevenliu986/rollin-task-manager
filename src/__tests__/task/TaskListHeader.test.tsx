import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import TaskListHeader from '../../components/task/TaskListHeader';

describe('TaskListHeader Component', () => {
    const mockOnTabChange = vi.fn();

    beforeEach(() => {
        mockOnTabChange.mockClear();
    });

    it('renders both tabs with correct initial active state', () => {
        render(
            <TaskListHeader
                activeTab="inProgress"
                onTabChange={mockOnTabChange}
            />
        );

        const inProgressTab = screen.getByRole('tab', { name: 'In Progress' });
        const completedTab = screen.getByRole('tab', { name: 'Completed' });

        expect(inProgressTab).toBeInTheDocument();
        expect(completedTab).toBeInTheDocument();

        expect(inProgressTab).toHaveAttribute('aria-selected', 'true');
        expect(completedTab).toHaveAttribute('aria-selected', 'false');

        expect(inProgressTab).toHaveStyle({ color: '#FFF532' });
        expect(completedTab).toHaveStyle({ color: '#FFF' });
    });

    it('shows indicator bar for active tab', () => {
        render(
            <TaskListHeader
                activeTab="inProgress"
                onTabChange={mockOnTabChange}
            />
        );

        const indicator = screen.getByRole('tab', { name: 'In Progress' })
            .querySelector('div[aria-hidden="true"]');

        expect(indicator).toBeInTheDocument();
        expect(indicator).toHaveStyle({
            backgroundColor: '#FFF532',
            height: '2px'
        });
    });

    it('calls onTabChange when clicking a tab', () => {
        render(
            <TaskListHeader
                activeTab="inProgress"
                onTabChange={mockOnTabChange}
            />
        );

        const completedTab = screen.getByRole('tab', { name: 'Completed' });
        fireEvent.click(completedTab);

        expect(mockOnTabChange).toHaveBeenCalledTimes(1);
        expect(mockOnTabChange).toHaveBeenCalledWith('completed');
    });

    it('updates UI when activeTab prop changes', () => {
        const { rerender } = render(
            <TaskListHeader
                activeTab="inProgress"
                onTabChange={mockOnTabChange}
            />
        );

        expect(screen.getByRole('tab', { name: 'In Progress' }))
            .toHaveAttribute('aria-selected', 'true');

        rerender(
            <TaskListHeader
                activeTab="completed"
                onTabChange={mockOnTabChange}
            />
        );

        expect(screen.getByRole('tab', { name: 'Completed' }))
            .toHaveAttribute('aria-selected', 'true');
        expect(screen.getByRole('tab', { name: 'In Progress' }))
            .toHaveAttribute('aria-selected', 'false');
    });

    it('has proper accessibility attributes', () => {
        render(
            <TaskListHeader
                activeTab="inProgress"
                onTabChange={mockOnTabChange}
            />
        );

        const tablist = screen.getByRole('tablist');
        expect(tablist).toHaveAttribute('aria-label', 'Task status tabs');

        const inProgressTab = screen.getByRole('tab', { name: 'In Progress' });
        expect(inProgressTab).toHaveAttribute('aria-controls', 'inProgress-tabpanel');

        const completedTab = screen.getByRole('tab', { name: 'Completed' });
        expect(completedTab).toHaveAttribute('aria-controls', 'completed-tabpanel');
    });

    it('matches snapshot for inProgress active tab', () => {
        const { asFragment } = render(
            <TaskListHeader
                activeTab="inProgress"
                onTabChange={mockOnTabChange}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot for completed active tab', () => {
        const { asFragment } = render(
            <TaskListHeader
                activeTab="completed"
                onTabChange={mockOnTabChange}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});