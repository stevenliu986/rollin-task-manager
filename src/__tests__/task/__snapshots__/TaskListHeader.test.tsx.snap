// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskListHeader Component > matches snapshot for completed active tab 1`] = `
<DocumentFragment>
  <div
    aria-label="Task status tabs"
    role="tablist"
    style="width: 100%; height: 53px; display: flex; justify-content: space-evenly; align-items: center; background-color: rgb(37, 37, 37); border-top-left-radius: 16px; border-top-right-radius: 16px;"
  >
    <button
      aria-controls="inProgress-tabpanel"
      aria-selected="false"
      role="tab"
      style="padding: 0px 25px; color: rgb(255, 255, 255); position: relative; cursor: pointer; background: none; font-family: inherit;"
    >
      In Progress
    </button>
    <button
      aria-controls="completed-tabpanel"
      aria-selected="true"
      role="tab"
      style="padding: 0px 25px; color: rgb(255, 245, 50); position: relative; cursor: pointer; background: none; font-family: inherit;"
    >
      Completed
      <div
        aria-hidden="true"
        style="position: absolute; bottom: -8px; left: 0px; width: 100%; height: 2px; background-color: rgb(255, 245, 50);"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`TaskListHeader Component > matches snapshot for inProgress active tab 1`] = `
<DocumentFragment>
  <div
    aria-label="Task status tabs"
    role="tablist"
    style="width: 100%; height: 53px; display: flex; justify-content: space-evenly; align-items: center; background-color: rgb(37, 37, 37); border-top-left-radius: 16px; border-top-right-radius: 16px;"
  >
    <button
      aria-controls="inProgress-tabpanel"
      aria-selected="true"
      role="tab"
      style="padding: 0px 25px; color: rgb(255, 245, 50); position: relative; cursor: pointer; background: none; font-family: inherit;"
    >
      In Progress
      <div
        aria-hidden="true"
        style="position: absolute; bottom: -8px; left: 0px; width: 100%; height: 2px; background-color: rgb(255, 245, 50);"
      />
    </button>
    <button
      aria-controls="completed-tabpanel"
      aria-selected="false"
      role="tab"
      style="padding: 0px 25px; color: rgb(255, 255, 255); position: relative; cursor: pointer; background: none; font-family: inherit;"
    >
      Completed
    </button>
  </div>
</DocumentFragment>
`;
